src/sentry/api/bases/avatar.py

```diff
@@ -40,8 +56,15 @@ def get_avatar_filename(self, obj):
         return f"{obj.id}.png"
 
     def put(self, request, **kwargs):
-        obj = kwargs[self.object_type]
-        serializer = AvatarSerializer(data=request.data, context=self.get_serializer_context(obj))
+        obj = kwargs.pop(self.object_type, None)
+        if self.object_type == "sentry_app":
+            serializer = SentryAppLogoSerializer(
+                data=request.data, context=self.get_serializer_context(obj)
+            )
+        else:
+            serializer = AvatarSerializer(
+                data=request.data, context=self.get_serializer_context(obj)
+            )
```

Comment:
nit: Can be simplified to a ternary if you want

Reply:
Agh I thought of that but then I thought it'd be kind of unreadable - let me see what it looks like after the linter has it's way.